File: backend/app/schemas/vocabulary.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime


class CharacterCreate(BaseModel):
    """录入生字请求"""
    character: str
    pinyin: Optional[str] = None
    radical: Optional[str] = None
    words: Optional[List[str]] = None
    example: Optional[str] = None
    brainstorm: Optional[List[str]] = None


class CharacterOut(BaseModel):
    """生字返回"""
    id: str
    character: str
    pinyin: Optional[str]
    radical: Optional[str]
    words: List[str]
    example: Optional[str]
    brainstorm: List[str]
    status: str
    review_count: int
    correct_count: int
    wrong_count: int
    group_id: int
    review_round: int
    last_reviewed_at: Optional[datetime]
    next_review_at: Optional[datetime]
    created_at: datetime

    class Config:
        from_attributes = True


class CharacterBatchCreate(BaseModel):
    """批量录入生字"""
    characters: List[str]


class QuizGroup(BaseModel):
    """检测分组返回"""
    group_id: int
    characters: List[CharacterOut]
    total: int


class ReviewSubmit(BaseModel):
    """提交检测结果"""
    character_id: str
    result: str  # know | unknow | study
    group_id: int
    review_round: int = 0


class ReviewSubmitBatch(BaseModel):
    """批量提交检测结果"""
    results: List[ReviewSubmit]


class ReadingRequest(BaseModel):
    """生成精读短文请求"""
    character_ids: Optional[List[str]] = None  # 指定生字，不传则取最近不认识的
    theme: Optional[str] = "日常"               # 主题


class ReadingOut(BaseModel):
    """精读短文返回"""
    title: str
    content: List[dict]  # [{"hz": "春", "py": "chūn", "highlight": false}, ...]
    summary: str

File: frontend/src/api/vocabulary.ts
import { request } from './request';
import { getStudentId } from '../utils/auth';

export interface Character {
  id: string;
  character: string;
  pinyin?: string;
  radical?: string;
  words: string[];
  example?: string;
  brainstorm: string[];
  status: string;
  review_count: number;
  correct_count: number;
  wrong_count: number;
  group_id: number;
  review_round: number;
  last_reviewed_at?: string;
  next_review_at?: string;
  created_at: string;
}

export interface QuizGroup {
  group_id: number;
  characters: Character[];
  total: number;
}

export interface ReviewResult {
  character_id: string;
  result: 'know' | 'unknow' | 'study';
  group_id: number;
  review_round: number;
}

export interface ReadingPassage {
  title: string;
  author?: string;
  content: { hz: string; py: string; highlight?: boolean }[];
  summary: string;
  highlight_words?: string[];
}

const studentId = getStudentId() || 'demo-student';

export async function addCharacter(character: string) {
  return request<Character>({
    url: `/students/${studentId}/literacy/characters`,
    method: 'POST',
    data: { character },
  });
}

export async function addCharactersBatch(characters: string[]) {
  return request<Character[]>({
    url: `/students/${studentId}/literacy/characters/batch`,
    method: 'POST',
    data: { characters },
  });
}

export async function getCharacters(status?: string) {
  return request<Character[]>({
    url: `/students/${studentId}/literacy/characters`,
    method: 'GET',
    data: status ? { status } : undefined,
  });
}

export async function getQuizGroup() {
  return request<QuizGroup>({
    url: `/students/${studentId}/literacy/quiz`,
    method: 'GET',
  });
}

export async function submitQuizResults(results: ReviewResult[]) {
  return request<Character[]>({
    url: `/students/${studentId}/literacy/quiz/submit`,
    method: 'POST',
    data: { results },
  });
}

export async function generateReading(theme?: string) {
  return request<ReadingPassage>({
    url: `/students/${studentId}/literacy/reading`,
    method: 'POST',
    data: { theme },
  });
}
